sdks/go/pkg/beam/registration/registration.tmpl

```diff
@@ -241,13 +242,320 @@ type teardown1x1 interface {
 	Teardown(ctx context.Context) error
 }
 
+type createAccumulator0x1[T any] interface {
+    CreateAccumulator() T
+}
+
+type createAccumulator0x2[T any] interface {
+    CreateAccumulator() (T, error)
+}
+
+type addInput2x1[T1, T2 any] interface {
+    AddInput(a T1, i T2) T1
+}
+
+type addInput2x2[T1, T2 any] interface {
+    AddInput(a T1, i T2) (T1, error)
+}
+
+type mergeAccumulators2x1[T any] interface {
+    MergeAccumulators(a0 T, a1 T) T
+}
+
+type mergeAccumulators2x2[T any] interface {
+    MergeAccumulators(a0 T, a1 T) (T, error)
+}
+
+type extractOutput1x1[T1, T2 any] interface {
+    ExtractOutput(a T1) T2
+}
+
+type extractOutput1x2[T1, T2 any] interface {
+    ExtractOutput(a T1) (T2, error)
+}
+
+{{range $accum := upto 3}}{{$genericParams := (add $accum 1)}}
+// Accumulator{{$genericParams}} registers an accumulator (CombineFn) DoFn's structural functions
```

Comment:
We probably want to call this Combiner1 etc.
They are called CombineFns or Combiners not accumulators. Accumulators are the intermediate type that CombineFns operate on. See https://beam.apache.org/documentation/programming-guide/#combine for the usual user nomenclature.

Eg. for a mean Combinefn, one could have (some number, say `int`) as AddInput's input type. AddInput would add it to the accumulator type `struct{Count int, Sum float64}`, MergeAccumulators merges them together, and ExtractOutput does the division of the Sum by the Count, to get the `float64` output type for the Mean.

Reply:
Yeah, that makes sense - updated!